docs/research/data_phases.md

```diff
@@ -0,0 +1,169 @@
+Phases of Information in Data Provider
+======================================
+
+## Background
+
+This document discusses the point in the app lifecycle at which specific inputs to i18n libraries are known, with implications on data layout and fallback behavior.
+
+## Three Phases of Information
+
+There are three points at which information may be known to the data provider:
+
+1. **Compile Time:** Based on the *programmer*. Can be determined by static analysis of code.
+2. **Construction Time:** Based on the *end user*. Can be determined from the locale, including Unicode extensions.
+3. **Format Time:** Based on the *runtime*. Can only be determined from the specific item being formatted (e.g., date or number).
+
+### Example: Duration Formatting
+
+Suppose you are formatting a time duration, such as a countdown timer for a video game. The breakdown of information known in each phase is:
+
+1. **Compile Time**
+    - Width (long, short, narrow, digital, …)
+2. **Construction Time**
+    - Locale
+    - Numbering system\*
+3. **Format Time**
+    - Specific set of time units (hour, minute, second, …)
+    - Plural form of the time units
+
+\* The numbering system determines the choice of digits. It does not currently affect the choice of unit pattern, but it may in the future.
+
+### Example: Date Formatting
+
+Date formatting is more complicated, but suppose you are formatting a date without a time based on a components bag. Information breakdown:
+
+1. **Compile Time**
+    - Coarse fields (months, weekdays, time zones, …)
+2. **Construction Time**
+    - Locale
+    - Calendar system\*
+    - Width\*\* (numeric, abbreviated, wide, …)
+3. **Format Time**
+    - Specific month, weekday, time zone, …
+
+\* The calendar system is special, because the program may or may not choose to support all calendar systems due to the amount of code and data that they require. Therefore, the calendar system may be known at compile time for certain applications.
+
+\*\* The width of fields can't be known until construction time because that might depend on the locale. For a given components bag, some locales may prefer wider month names whereas others may prefer narrower ones.
+
+## Resource Paths
+
+The phase in which information is known should influence whether that information ends up in the resource key, resource options, or data struct. As a general rule of thumb:
```

Comment:
Suggestion (optional): This is the first use of resource key, resource options, and data struct as terms in this document, but they haven't been defined yet. I was initially somewhat unsure as to what they meant precisely, until I read further below with the examples. It might be nice to hint to the reader (through words, or a hyperlink) that the definitions are below (or maybe discussed more in depth in other files.)